staging/src/k8s.io/client-go/discovery/fake/discovery.go

```diff
@@ -77,7 +79,9 @@ func (c *FakeDiscovery) ServerGroupsAndResources() ([]*metav1.APIGroup, []*metav
 		Verb:     "get",
 		Resource: schema.GroupVersionResource{Resource: "resource"},
 	}
-	c.Invokes(action, nil)
+	if _, err = c.Invokes(action, nil); err != nil {
+		return nil, nil, err
```

Comment:
ServerGroupsAndResources can return partial results. Returning nil, nil seems wrong.

Reply:
@sttts Thanks for the review. Updated. PTAL.